rpm/0002-Add-D-Bus-interface-for-querying-client-process-deta.patch

```diff
@@ -0,0 +1,207 @@
+From 94bc51efc349ac54889d2af5db8ace6306e0dec0 Mon Sep 17 00:00:00 2001
+From: Simo Piiroinen <simo.piiroinen@jolla.com>
+Date: Thu, 16 Sep 2021 12:18:41 +0300
+Subject: [PATCH] Add D-Bus interface for querying client process details
+
+Various services in Sailfish OS need to find out details about D-Bus
+clients e.g. for purpose the of showing originating application of
+notifications. When dealing with sandboxed applications launched via
+sailjail daemon, use of standard D-Bus client identification methods
+such as org.freedesktop.DBus.GetConnectionUnixProcessID yields pid of
+the xdg-dbus-proxy process rather than the application behind the proxy.
+
+Make it so that each bus facing connection xdg-dbus-proxy implicitly
+provides a D-Bus interface that can be used for querying details about
+the client behind the proxy connection.
+
+Implement org.sailfishos.sailjailed.Identify() method call that returns
+pid, uid, and other details about the connected client.
+
+D-Bus policy configuration allowing/denying such method calls to be made
+is assumed to be defined elsewhere.
+
+Signed-off-by: Simo Piiroinen <simo.piiroinen@jolla.com>
+---
+ flatpak-proxy.c | 138 +++++++++++++++++++++++++++++++++++++++++++++++-
+ 1 file changed, 136 insertions(+), 2 deletions(-)
+
+diff --git a/flatpak-proxy.c b/flatpak-proxy.c
+index 8bf9a4f..4ca3223 100644
+--- a/flatpak-proxy.c
++++ b/flatpak-proxy.c
+@@ -22,6 +22,7 @@
+ 
+ #include <unistd.h>
+ #include <string.h>
++#include <stdio.h>
+ 
+ #include "flatpak-proxy.h"
+ 
+@@ -1998,7 +1999,7 @@ update_socket_messages (ProxySide *side, Buffer *buffer, Header *header)
+ }
+ 
+ static void
+-queue_fake_message (FlatpakProxyClient *client, GDBusMessage *message, ExpectedReplyType reply_type)
++queue_bus_message (FlatpakProxyClient *client, GDBusMessage *message)
+ {
+   Buffer *buffer;
+ 
+@@ -2009,6 +2010,12 @@ queue_fake_message (FlatpakProxyClient *client, GDBusMessage *message, ExpectedR
+   g_object_unref (message);
+ 
+   queue_outgoing_buffer (&client->bus_side, buffer);
++}
++
++static void
++queue_fake_message (FlatpakProxyClient *client, GDBusMessage *message, ExpectedReplyType reply_type)
++{
++  queue_bus_message(client, message);
+   queue_expected_reply (&client->client_side, client->last_serial, reply_type);
+ }
+ 
+@@ -2306,6 +2313,129 @@ handle_deny:
+     queue_initial_name_ops (client);
+ }
+ 
++static gboolean
++handle_proxy_method(FlatpakProxyClient *client, Header *header, Buffer *buffer)
++{
++  if (header->type != G_DBUS_MESSAGE_TYPE_METHOD_CALL)
++    return FALSE;
++
++  if (g_strcmp0 (header->interface, "org.sailfishos.sailjailed"))
++    return FALSE;
++
++  GDBusMessage *message = g_dbus_message_new_from_blob (buffer->data, buffer->size,
++                                                        G_DBUS_CAPABILITY_FLAGS_NONE, NULL);
++  if (message)
```

Comment:
Reduce nesting somehow.

```
if (!message)
  return TRUE;
```

Reply:
Having "not for us to handle" cases use early return FALSE and otherwise have a single return TRUE attempts to make the intent of the code obvious. And at least to me adding early return TRUE would make things less clear.

Now that the inner parts are simpler, does the nesting level still bother you?

Reply:
It's better now, but anyway it looks strange for me, because there are 7 nested levels.

Maybe it will be better to left check `!g_strcmp0(header->member, "Identify"))` and move another checks to the another function `get_pid(FlatpakProxyClient *client)` ?

Reply:
I decided to pretend that there might be more of these later on and split it into: handle_proxy_methods() and handle_proxy_identify_method(). That has logical structure to it and brings the deepest nesting level down a bit.